### PR TITLE
Core/Battleground: fix logic missmatch in bg IoC

### LINKED ISSUE #2593: Isle of Conquest Scoreboard Error
##### SMALL DESCRIPTION:

Joined BG Isle of Conquest with one pleayer each faction, 2 Horde are shown. 

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Scoreboard should work like all other scorebnoards and show the players each faction correctly. 
Example: 

image

##### CURRENT BEHAVIOUR:

The error appears only on Random Queue, not Specific Queue! 

**Scoreboard from Random Queue:** 

image

**Scoreboard from Specific Queue:**

image

##### STEPS TO REPRODUCE THE PROBLEM:

1. create 1 H/A char on different accounts
2. .debug bg
3. Join via Random Queue the BG Isle of Conquest and open Scoreboard

> You can remove the "Deserteur" aura with .unaura 26013 

##### EXTRA NOTES:

[Server.log](https://github.com/azerothcore/azerothcore-wotlk/files/4081695/Server.log)

##### BRANCH(ES):

master

##### AC HASH/COMMIT:

bb42c7c5eb03+

##### OPERATING SYSTEM:

Windows 10 pro 64 bit

##### MODULES:

- Auctionhouse Bot
- SoloLFG
- Autobalance

##### OTHER CUSTOMIZATIONS:

- Max skill on levelup

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/87195215-isle-of-conquest-scoreboard-error?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Viste**
## ISSUES ADDRESSED:
- must Closes #2593

## Target branch(es):
- [x] Master

### CODE REVIEW

File: src/server/game/Battlegrounds/Zones/BattlegroundIC.cpp [hunk 1]
```diff
@@ -352,7 +352,7 @@ void BattlegroundIC::HandleAreaTrigger(Player* player, uint32 trigger)
 
 void BattlegroundIC::UpdatePlayerScore(Player* player, uint32 type, uint32 value, bool doAddHonor)
 {
-    std::map<uint64, BattlegroundScore*>::iterator itr = PlayerScores.find(player->GetGUID());
+    BattlegroundScoreMap::iterator itr = PlayerScores.find(player->GetGUID());
```
Comment (Winfidonarleyan) [inline src/server/game/Battlegrounds/Zones/BattlegroundIC.cpp]: Why not use?
```cpp 
auto const&
```

Comment (Viste) [reply src/server/game/Battlegrounds/Zones/BattlegroundIC.cpp]: потому что если я начну делать нормально, я так просто не остановлюсь.

Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/Zones/BattlegroundIC.cpp]: велком ту вархэд 🗡

**Comment by Winfidonarleyan** [comment]
I think waiting for testing is stupid. Merge this.

**Comment by FrancescoBorzi** [comment]
@Winfidonarleyan for such a small change, if you tested it, it's fine.

**Pull request merged**